### ISSUE TITLE
False positive for `try-except-raise` with multiple exceptions catching

### ISSUE THREAD

**Issue by fly** [PyCQA/pylint#8051]
### Bug description

Error is produced when a single `except` statement catches several exceptions for re-raising **and** when these exceptions are not in the current namespace.

Example (both `Exception1` and `Exception2` here are inherited directly from `Exception`):
```python
from app import errors

def check():
    try:
        return do_something()
    except (errors.Exception1, errors.Exception2):  # The except handler raises immediately (try-except-raise)
        raise
    except Exception as e:
        raise errors.Exception1("Boom!") from e
```

Note that if I bring exceptions directly into namespace, then it works:
```python
from app.errors import Exception1, Exception2

def check():
    try:
        return do_something()
    except (Exception1, Exception2): 
        raise
    except Exception as e:
        raise Exception1("Boom!") from e
```

It also works if I use separate except statements for each:

```python
from app import errors

def check():
    try:
        return do_something()
    except errors.Exception1:
        raise
    except errors.Exception2:
        raise
    except Exception as e:
        raise errors.Exception1("Boom!") from e
```

### Command used

```shell
pylint check.py
```

### Pylint output

```shell
************* Module check
check.py:19:4: W0706: The except handler raises immediately (try-except-raise)
```

### Expected behavior

No errors

### Pylint version

```shell
pylint 2.15.10
astroid 2.13.2
Python 3.9.16 (main, Dec  7 2022, 01:12:08) 
[GCC 11.3.0]
```

**Comment by Pierre-Sassoulas** [comment]
Hello, thank you for opening the issue. What is the inheritance tree of ``errors.Exception1``, ``errors.Exception2`` and ``Exception`` ?

**Comment by fly** [comment]
both `Exception1` and `Exception2` here are inherited directly from `Exception`. `Exception` is built in

**Comment by Pierre-Sassoulas** [comment]
It's indeed a false positive then. We probably need to use inference to handle exception's aliases.
